network/src/noise/fuzzing.rs

```diff
@@ -87,35 +87,66 @@ impl AsyncRead for ExposingSocket {
 // Actual code to generate corpus
 //
 
-// let's cache the deterministic keypair
-pub static KEYPAIR: Lazy<(x25519::PrivateKey, x25519::PublicKey)> = Lazy::new(|| {
+// cache the deterministic keypairs
+pub static KEYPAIRS: Lazy<(
+    (x25519::PrivateKey, x25519::PublicKey, PeerId),
+    (x25519::PrivateKey, x25519::PublicKey, PeerId),
+)> = Lazy::new(|| {
     let mut rng = ::rand::rngs::StdRng::from_seed(TEST_SEED);
-    let private_key = x25519::PrivateKey::generate(&mut rng);
-    let public_key = private_key.public_key();
-    (private_key, public_key)
+
+    let initiator_key = x25519::PrivateKey::generate(&mut rng);
+    let initiator_public_key = initiator_key.public_key();
+    let initiator_peer_id = PeerId::from_identity_public_key(initiator_public_key);
+
+    let responder_key = x25519::PrivateKey::generate(&mut rng);
+    let responder_public_key = responder_key.public_key();
+    let responder_peer_id = PeerId::from_identity_public_key(responder_public_key);
+
+    (
+        (initiator_key, initiator_public_key, initiator_peer_id),
+        (responder_key, responder_public_key, responder_peer_id),
+    )
 });
 
 fn generate_first_two_messages() -> (Vec<u8>, Vec<u8>) {
     // build
-    let (private_key, public_key) = KEYPAIR.clone();
-    let peer_id = PeerId::from_identity_public_key(public_key);
-    let initiator = NoiseUpgrader::new(peer_id, private_key.clone(), HandshakeAuthMode::ServerOnly);
-    let responder = NoiseUpgrader::new(peer_id, private_key, HandshakeAuthMode::ServerOnly);
+    let (
+        (initiator_key, initiator_public_key, initiator_peer_id),
+        (responder_key, responder_public_key, responder_peer_id),
+    ) = KEYPAIRS.clone();
+
+    let initiator = NoiseUpgrader::new(
+        initiator_peer_id,
+        initiator_key,
+        HandshakeAuthMode::ServerOnly,
+    );
+    let responder = NoiseUpgrader::new(
+        responder_peer_id,
+        responder_key,
+        HandshakeAuthMode::ServerOnly,
+    );
 
     // create exposing socket
-    let (dialer_socket, listener_socket) = ExposingSocket::new_pair();
+    let (initiator_socket, listener_socket) = ExposingSocket::new_pair();
```

Comment:
I like `initiator` better than `dialer`

Reply:
woops! my find and replace grabbed a bit too much...